File: python/ch_1_simple.py
#! /usr/bin/env python
# coding:utf8

import os

def main():
    # all codes before fork will be calles only once
    print("main start...")

    pid = os.fork()
    print("this point for child process is 'resume'-ing running, while for parent procss is 'continue'-ing running")
    if pid == 0:
        # Codes running in the child process eventually.
        print("%d (child) just was created by %d." % (os.getpid(), os.getppid()))
    else:
        # Codes running in the parent process eventually.
        print("%d (parent) just created %d." % (os.getpid(), pid))

    print("main end...")

if __name__ == "__main__":
    main()

File: php/ch_1_simple.php
#! /usr/bin/env php
<?php

function main() {
    print("main start...\n");

    $pid = pcntl_fork();
    print("this point for child process is 'resume'-ing running, while for parent procss is 'continue'-ing running\n");
    if ($pid === 0) {
        // Codes running in the child process eventually.
        printf("%d (child) just was created by %d.\n", getmypid(), posix_getppid());
    } else {
        // Codes running in the parent process eventually.
        printf("%d (parent) just created %d.\n", getmypid(), $pid);
    }

    print("main end...\n");
}

main();
